packages/relay/src/lib/services/cacheService/cacheService.ts

```diff
@@ -0,0 +1,187 @@
+   * the internal cache is used as a fallback.
+   * @param {string} key - The key to associate with the value.
+   * @param {*} value - The value to cache.
+   * @param {string} callingMethod - The name of the method calling the cache.
+   * @param {number} ttl - Time to live for the cached value in milliseconds (optional).
+   * @param {string} requestIdPrefix - A prefix to include in log messages (optional).
+   * @param {boolean} shared - Whether to use the shared cache (optional, default: false).
+   */
+  public set(
+    key: string,
+    value: any,
+    callingMethod: string,
+    ttl?: number,
+    requestIdPrefix?: string,
+    shared: boolean = false
+  ): void {
+    if (shared && this.isSharedCacheEnabled) {
+      try {
+        return this.sharedCache.set(key, value, callingMethod, ttl, requestIdPrefix);
+      } catch (error) {
+        const redisError = new RedisCacheError(error);
+        this.logger.error(`Error occured while setting the cache to Redis. Fallback to internal cache. Error is: ${redisError.fullError}`);
+      }
+      this.internalCache.set(key, value, callingMethod, ttl, requestIdPrefix);
+    } else {
+      this.internalCache.set(key, value, callingMethod, ttl, requestIdPrefix);
+    }
+  }
+
+  /**
+   * Deletes a cached value associated with the given key.
+   * If the shared cache is enabled and an error occurs while deleting from it, just logs the error.
+   * Else the internal cache deletion is attempted.
+   * @param {string} key - The key associated with the cached value to delete.
+   * @param {string} callingMethod - The name of the method calling the cache.
+   * @param {string} requestIdPrefix - A prefix to include in log messages (optional).
+   * @param {boolean} shared - Whether to use the shared cache (optional, default: false).
+   */
+  public delete(key: string, callingMethod: string, requestIdPrefix?: string, shared: boolean = false): void {
+    if (shared && this.isSharedCacheEnabled) {
+      try {
+        return this.sharedCache.delete(key, callingMethod, requestIdPrefix);
+      } catch (error) {
+        const redisError = new RedisCacheError(error);
+        this.logger.error(`Error occured while deleting cache from Redis. Error is: ${redisError.fullError}`);
+      }
+    } else {
+      this.internalCache.delete(key, callingMethod, requestIdPrefix);
+    }
+  }
+
+  /**
+   * Clears the cache.
+   * If the shared cache is enabled and an error occurs while clearing it, just logs the error.
+   * Else the internal cache clearing is attempted.
+   * @param {boolean} shared - Whether to clear the shared cache (optional, default: false).
+   */
+  public clear(shared: boolean = false): void {
+    if (shared && this.isSharedCacheEnabled) {
+      try {
+        return this.sharedCache.clear();
+      } catch (error) {
+        const redisError = new RedisCacheError(error);
+        this.logger.error(`Error occured while clearing Redis cache. Error is: ${redisError.fullError}`);
```

Comment:
All the logs in this class as missing the use of the `requestIdPrefix` values
```suggestion
        this.logger.error(`${requestIdPrefix} Error occurred while clearing Redis cache. Error is: ${redisError.fullError}`);
```

Reply:
Added and to other missing places as well.